### PR TITLE
Buy, Sell, Leave button size fix

### PR DESCRIPTION
**Pull request by ChamEV**
Adjusted the clickable and hover area of the buy, sell, and leave buttons in the shipyard and outfit shops such that it matches the visual button area.

### CODE REVIEW

File: source/ShopPanel.cpp [hunk 1]
```diff
@@ -1157,11 +1157,11 @@ char ShopPanel::CheckButton(int x, int y)
 		return ' ';
 	
 	x -= Screen::Right() - SIDE_WIDTH;
-	if(x < 80)
+	if(x > 9 && x < 69)
```
[the remark this answers is not in the document]
Comment (ChamEV) [reply source/ShopPanel.cpp]: That's what I tried initially, but it just looked off.  I think there was a bit of optical illusion with the mouse and button edge when I did it that way.  Opened a screenshot in gimp and the right pixel is off by 1.  The left pixel is perfect.  The area listed above is actually 59 pixels wide, not 58.

**Comment by ChamEV** [comment]
Should be perfect now.

**Comment by petervdmeer** [comment]
> Should be perfect now.

Great!

**Comment by Pointedstick** [comment]
I get why you did this, but I think the expanded hit areas may be intentional. Many small UI elements incorporate expanded hit areas like this to aid in quickly clicking on them. I almost wonder if we should do the opposite and extend the hit areas all the way to the bottom of the screen to take advantage of Fitts' Law.

**Comment by ChamEV** [comment]
It's hard to tell if it's intentional.  The accuracy of the click zones vs the button sizes is different for every set of buttons.  The buy all/sell all buttons on the trade panel are precise, but the [buy] and [sell] buttons are anything but.

The buy click zone in this extends all the way to 1-2 pixels to the left of [sell] for instance.
image

There are a lot of areas where precise clicking on much smaller things is required compared to these large buttons.

Also of note is these button definitions may be replaced by definitions in interfaces.txt like the map buttons, though I'm not sure of the progress on that or exactly how to do it at this time.

**Pull request merged**